tests/test_facebook_feed_block.py

```diff
@@ -78,3 +81,56 @@ def prepare_url(blk, feed_type):
         blk.feed_type = FeedType.PROMOTABLE_POSTS
         blk._prepare_url()
         self.assertEqual(blk.url, prepare_url(blk, 'promotable_posts'))
+
+    @patch.object(RESTPolling, "_retry")
+    @patch.object(FacebookFeed, "_authenticate")
+    @patch("requests.get")
+    def test_username_query(self, mock_get, mock_auth, mock_retry):
+        """ username queries get a 404 from Facebook """
+        blk = FacebookFeed()
+        self.configure_block(blk, {
+            "queries": [
+                "username1",
+                "username2"
+            ]
+        })
+        mock_get.return_value = MagicMock()
+        mock_get.return_value.status_code = 404
+        mock_get.return_value.json.return_value = \
+            {'error':
+             {'message':
+              '(#803) Cannot query users by their username (username1)',
+              'code': 803,
+              'type': 'OAuthException'
+              }
+             }
+        self.assertEqual(0, blk._idx)
+        blk.poll()
+        # skip to next idx because we are not retrying.
+        self.assertEqual(1, blk._idx)
+
+    @patch.object(RESTPolling, "_retry")
+    @patch.object(FacebookFeed, "_authenticate")
+    @patch("requests.get")
+    def test_retry(self, mock_get, mock_auth, mock_retry):
+        """ username queries get a 404 from Facebook """
```

Comment:
Is this testing a normal 404 for usernames? It looks like it's testing if some other type of error occurs